src/librustdoc/html/static/main.js

```diff
@@ -2241,8 +2241,7 @@ function defocusSearchBar() {
                 relatedDoc = relatedDoc.nextElementSibling;
             }
 
-            if ((!relatedDoc && hasClass(docblock, "docblock") === false) ||
-                (pageId && document.getElementById(pageId))) {
+            if (!relatedDoc && hasClass(docblock, "docblock") === false) {
```

Comment:
Little note about this change for the reviewers: it was to prevent an element targetted by the url hash (so `#variant.X` for example) to be hidden. However, it has been long fixed and this code wasn't removed. And this case, it was a big issue in case you had `#implementors` as page id because then it prevented the impl blocks to be collapsed. I tested without this check for all other cases and it works fine.